applications/CompressiblePotentialFlowApplication/custom_processes/apply_far_field_process.cpp

```diff
@@ -0,0 +1,129 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Iñigo Lopez and Marc Núñez
+//
+
+// Project includes
+#include "apply_far_field_process.h"
+
+namespace Kratos {
+
+// Constructor for ApplyFarFieldProcess Process
+ApplyFarFieldProcess::ApplyFarFieldProcess(ModelPart& rModelPart, const double rReferencePotential)
+    : Process(), mrModelPart(rModelPart), mReferencePotential(rReferencePotential)
+{
+    mFreeStreamVelocity = mrModelPart.GetProcessInfo()[FREE_STREAM_VELOCITY];
+}
+
+void ApplyFarFieldProcess::Execute()
+{
+    FindFarthestUpstreamBoundaryNode();
+    AssignFarFieldBoundaryConditions();
+}
+
+void ApplyFarFieldProcess::FindFarthestUpstreamBoundaryNode()
+{
+    // Declaring omp variables, generating vectors of size = num_threads
+    std::size_t num_threads = omp_get_max_threads();
+    std::vector<double> min_projections(num_threads, std::numeric_limits<double>::epsilon());
+    std::vector<std::size_t> nodes_id_list(num_threads, 0);
+
+    //Find minimum node in the direction of the free stream velocity in openmp.
+    double distance_projection=0.0;
+    #pragma omp parallel for firstprivate (distance_projection)
+    for (int i = 0; i < static_cast<int>(mrModelPart.Nodes().size()); i++) {
+        auto it_node = mrModelPart.NodesBegin() + i;
+        std::size_t thread_id = omp_get_thread_num();
+        const auto& r_coordinates = it_node->Coordinates();
+        distance_projection = inner_prod(r_coordinates, mFreeStreamVelocity);
+        if (distance_projection < min_projections[thread_id]){
+            min_projections[thread_id] = distance_projection;
+            nodes_id_list[thread_id] = it_node->Id();
+        }
+    }
+
+    // Find minimum across all threads
+    std::size_t minimum_node_thread_id = 0;
+    for (std::size_t i_thread = 0; i_thread<num_threads; i_thread++){
+        if (min_projections[i_thread] < min_projections[minimum_node_thread_id]){
+            minimum_node_thread_id = i_thread;
+        }
+    }
+
+    // Storing final result in member variable
+    mpReferenceNode = mrModelPart.pGetNode(nodes_id_list[minimum_node_thread_id]);
+}
+
+void ApplyFarFieldProcess::AssignFarFieldBoundaryConditions()
+{
+    double velocity_projection = 0.0;
```

Comment:
Codacy is complaining about this variable being defined here and not inside the for loop.

If i understood correctly, it is better to declare these variables outside the for loop right? @roigcarlo (so can we ignore codacy?)

Reply:
According to my experience in this case is better. But now you make me doubt, give me 10 to make some research.

Reply:
Ok for this particular case maybe better inside

Reply:
Btw does this also applies to array_1d? Also inside? (Because codacy only complains about doubles)

Maybe its different for a different type of variable.

Reply:
Probably codacy dosn't understand what is an array_1d tbh.

Reply:
> Codacy is complaining about this variable being defined here and not inside the for loop.
> 
> If i understood correctly, it is better to declare these variables outside the for loop right?

I had the same situation while migrating the wake process. As far as I know it is better to declare the variables inside the for loop, the compiler typically optimizes better than us and it restricts the scope of the variable to inside the loop.